From e43d3adb9580a5aa7d634075bc496cbe80c29bcf Mon Sep 17 00:00:00 2001
From: Marek Marczykowski <marmarek@invisiblethingslab.com>
Date: Fri, 19 Oct 2018 08:02:13 +0200
Subject: [PATCH] avoid adding duplicated kernel entries

List kernel versions without duplicates, even when there are multiple
files related to the same kernel version.
Duplicated kernel versions here caused regenerating initramfs multiple
times and duplicated entries in xen.cfg.

QubesOS/qubes-issues#3624
---
 pyanaconda/payload/__init__.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pyanaconda/payload/__init__.py b/pyanaconda/payload/__init__.py
index 3b2cc6db6..0486df8c9 100644
--- a/pyanaconda/payload/__init__.py
+++ b/pyanaconda/payload/__init__.py
@@ -1113,7 +1113,7 @@ class PackagePayload(Payload):
                 if fnmatch(f, "/boot/vmlinuz-*") or
                    fnmatch(f, "/boot/efi/EFI/%s/vmlinuz-*" % self.instclass.efi_dir)))
 
-        return sorted(files, key=functools.cmp_to_key(versionCmp))
+        return sorted(set(files), key=functools.cmp_to_key(versionCmp))
 
     @property
     def rpmMacros(self):
-- 
2.17.2

